feat: narrow sidebar UI/UX

yikZero opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a localized shortcut tooltip for expanding the sidebar.

* **Improvements**
  * Enhanced sidebar collapse/expand UI with a hover-driven toggle, consistent label widths, and refined spacing.
  * Adjusted collapsed tab sizing, header horizontal padding, and bottom menu spacing for finer layout alignment.
  * Tweaked main window titlebar control positioning for improved alignment.
  * Added extra-small font variant tokens for compact text.

* **Other**
  * Simplified device-management tab translation handling.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yikZero committed:
feat: update translations for sidebar collapse functionality and add new device translation keys

- Updated translation keys for collapsing the sidebar to improve clarity in user interface.
- Added new translation key for "device" to enhance localization support across multiple languages.
- Ensured consistency in translation keys for better user experience.

yikZero committed:
Merge branch 'OneKeyHQ:x' into sidebar

yikZero committed:
feat: enhance DesktopLeftSideBar with hover effects and layout adjustments

- Added hover state management for the sidebar toggle to improve user interaction.
- Adjusted gap and text properties for better visual consistency in collapsed state.
- Refactored IconButton to YStack for improved accessibility and styling.
- Implemented onHoverIn and onHoverOut events to manage sidebar visibility more effectively.

yikZero committed:
fix: adjust layout and styling in desktop components

- Modified traffic light position in the main window for better alignment.
- Enhanced DesktopLeftSideBar with MotiView for improved hover effects and accessibility.
- Updated padding in BottomMenu for consistent spacing.
- Refined tab bar item styles in DesktopCustomTabBarItem for better visual consistency.
- Adjusted section header padding in DesktopCustomTabBar to accommodate collapsed state.

yikZero committed:
feat: enhance DesktopLeftSideBar with tooltip for sidebar expansion

- Integrated Tooltip component to provide user guidance for expanding the sidebar.
- Added new translation key for "expand sidebar" to improve localization support.
- Updated DesktopLeftSideBar to utilize the new tooltip for better user experience and accessibility.

yikZero committed:
fix: adjust SizableText width in DesktopCustomTabBarItem for better layout

- Updated the width property of SizableText in DesktopCustomTabBarItem to ensure consistent styling and alignment.
- This change enhances the visual consistency of the tab bar items in the desktop interface.

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/27791267-31e0-4cb5-b172-19fbc7794935/license?refs=3faa33d1444c2732273ccb54f0313f4182ec5ffc&source=prChecksComment) |
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/27791267-31e0-4cb5-b172-19fbc7794935?refs=3faa33d1444c2732273ccb54f0313f4182ec5ffc&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

yikZero committed:
Merge branch 'x' into sidebar

yikZero committed:
fix: adjust padding in DesktopLeftSideBar for improved layout

- Updated top and bottom padding in TabItemView to enhance visual consistency in the sidebar's collapsed state.
- This change aims to improve the overall user experience by refining the layout of tab items.

yikZero committed:
fix: update font variants and padding in Tamagui configuration

- Added new font sizes and weights for bodyXs and bodyXsMedium to enhance text styling consistency.
- Adjusted padding in TabItemView for improved layout in the DesktopLeftSideBar, ensuring better visual alignment in the collapsed state.
- These changes aim to refine the overall user experience by enhancing the visual consistency of components.

yikZero commented on packages/kit/src/views/Discovery/components/DesktopCustomTabBarItem/index.tsx:
```diff
@@ -293,13 +298,12 @@ function DesktopCustomTabBarItem({
           >
             <DesktopTabItemImage avatarSrc={tab?.favicon} selected={isActive} />
-            <SizableText size="$bodyMd" numberOfLines={1}>
+            <SizableText size="$bodyMd" width="$24" numberOfLines={1}>
               {label}
```
为了关闭按钮能够在相同的位置，需控制 tooltip 宽度一致。

yikZero committed:
refactor: simplify hover state management in DesktopLeftSideBar

- Removed unnecessary state and effects related to toggle hover management to streamline the component.
- Adjusted styling properties for better visual alignment and hover effects.
- These changes aim to enhance the user experience by improving the performance and maintainability of the sidebar component.

yikZero committed:
Merge branch 'x' into sidebar

huhuanming committed:
Merge branch 'x' into sidebar
